third_party/terraform/tests/resource_storage_transfer_job_test.go

```diff
@@ -0,0 +1,165 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccStorageTransferJob_basic(t *testing.T) {
+	t.Parallel()
+
+	testDataSourceId := acctest.RandString(10)
+	testDataSinkId := acctest.RandString(10)
+	testTransferJobId := acctest.RandString(10)
+	testUpdatedDataSourceId := acctest.RandString(10)
+	testUpdatedDataSinkId := acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccStorageTransferJobDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccStorageTransferJob_basic(getTestProjectFromEnv(), testDataSourceId, testDataSinkId, testTransferJobId),
+			},
+			{
+				ResourceName:      "google_storage_transfer_job.transfer_job",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccStorageTransferJob_basic(getTestProjectFromEnv(), testDataSourceId, testUpdatedDataSinkId, testTransferJobId),
+				Check:  resource.TestCheckResourceAttr("google_storage_transfer_job.transfer_job.transfer_spec.0.gcs_data_sink.0", "bucket_name", testUpdatedDataSinkId),
+			},
+			{
+				ResourceName:      "google_storage_transfer_job.transfer_job",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccStorageTransferJob_basic(getTestProjectFromEnv(), testUpdatedDataSourceId, testUpdatedDataSinkId, testTransferJobId),
+				Check:  resource.TestCheckResourceAttr("google_storage_transfer_job.transfer_job.transfer_spec.0.gcs_data_source.0", "bucket_name", testUpdatedDataSinkId),
+			},
+			{
+				ResourceName:      "google_storage_transfer_job.transfer_job",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccStorageTransferJob_basic(project string, dataSourceId string, dataSinkId string, transferJobId string) string {
+	return fmt.Sprintf(`
+data "google_storage_transfer_project_service_account" "default" {
+  project       = "%s"
+}
+
+resource "google_storage_bucket" "data_source" {
+  name          = "test-data-source-bucket-%s"
+  project       = "%s"
+  force_destroy = true
+}
+
+resource "google_storage_bucket_iam_member" "data_source" {
+  bucket        = "${google_storage_bucket.data_source.name}"
+  role          = "roles/storage.admin"
+  member        = "serviceAccount:${data.google_storage_transfer_project_service_account.default.email}"
+
+  depends_on    = [
```

Comment:
This resource will already depend on those other resources because of the interpolations, so this block is redundant.